docs/reST/ref/math.rst

```diff
@@ -61,6 +61,38 @@ Multiple coordinates can be set using slices or swizzling
 
    .. ## math.clamp ##
 
+.. function:: lerp
+
+   | :sl:`interpolates between two values by a percentage.`
+   | :sg:`lerp(a, b, percent) -> float`
+
+   Linearly interpolates between ``a`` and ``b`` by ``percent`` using the formula ``a + (b-a) * percent``.
+   
+   If ``percent`` is ``0.5``, ``lerp`` will return the value half-way between ``a``
+   and ``b``. When ``a = 10`` and ``b = 20``, ``lerp(a, b, 0.5)`` will return ``15``.
+
+   ``lerp`` can be used for many things. You could rotate a sprite by a percentage with
+   ``angle = lerp(0, 360, percent)``. You could even scale an enemy's attack value
+   based on the level you're playing:
+
```

Comment:
I like the first part of this doc, I think mentioning the formula it uses clears up a lot of things...

The latter part of this docs is a bit specific and subjective, docs generally don't need to mention too many usage details (although in some cases they make sense and explain stuff better)

I also think that the doc keyword name "percent" could be renamed, because "percent" as a name gives the impression that this has to be between 0 and 100

Reply:
@MyreMylar said this on my issue: "Though more attention might need to be put on things like documentation and examples to introduce the concept of lerping in games to an audience who might never have heard of it. 'Linear interpolation' sounds like a scary thing that is actually simple and very useful (health bars spring to mind)." 

What do you think a good name would be for percent? Godot using lerp(from, to, weight).

Reply:
I like "weight" as a name

Apart from that naming, the rest of the docs looks good to go to me